src/aggregate/aggregate_exec.c

```diff
@@ -329,7 +332,9 @@ static void cursorRead(RedisModuleCtx *ctx, uint64_t cid, size_t count) {
     RedisModule_ReplyWithError(ctx, "Cursor not found");
     return;
   }
+  QueryError status = {0};
   AREQ *req = cursor->execState;
+  req->qiter.err = &status;
```

Comment:
hrm i feel this should be set deeper in the stack

Reply:
Deeper function in the stack share other code paths that already have this value set, this is the deeper it will get I am afraid .. (Unless we want to check if its NULL and only then set it but I less like it ...)